 Broadcast networkstatechange indication on exit from ScanModeState

WiFiStateMachine enters in to the ScanModeState, when the user Turns OFF
Wi-Fi with the location service enabled. When the user Turns ON Wi-Fi
again, the previous connected SSID( before the Turn OFF ) is still
displayed as Connected for a short duration. This is because of the delay
in issuing the network state change indication from the framework. This
indication currently happens only on receiving the state change indication
from the wpa_supplicant. (The Turn ON if happens after an immediate SCAN
request would result in a delay in the state change indication from
wpa_supplicant, as this shall be issued after the scan is done). Thus, to
avoid this issue of showing the network as connected for a shorter
duration, issue a network state change indication when the ScanModeState is
exited due to a Turn ON request.

diff --git a/service/java/com/android/server/wifi/WifiStateMachine.java b/service/java/com/android/server/wifi/WifiStateMachine.java
index 5b285f5..b0ea785 100644
--- a/service/java/com/android/server/wifi/WifiStateMachine.java
+++ b/service/java/com/android/server/wifi/WifiStateMachine.java
@@ -5286,6 +5286,7 @@ public boolean processMessage(Message message) {
                                         WifiConfiguration.INVALID_NETWORK_ID);
 
                         mOperationalMode = CONNECT_MODE;
+                        setNetworkDetailedState(DetailedState.DISCONNECTED);
                         transitionTo(mDisconnectedState);
                     } else {
                         // Nothing to do
